nuxeo-features/nuxeo-platform-rendition/nuxeo-platform-rendition-core/src/main/java/org/nuxeo/ecm/platform/rendition/service/RenditionServiceImpl.java

```diff
@@ -532,77 +533,63 @@ public Rendition getDefaultRendition(DocumentModel doc, String reason, Map<Strin
-        for (int i = defaultRenditionDescriptors.size() - 1; i >= 0; i--) {
-            DefaultRenditionDescriptor d = defaultRenditionDescriptors.get(i);
-            if ((StringUtils.isEmpty(reason) && StringUtils.isEmpty(d.reason))
-                    || (reason != null && reason.equals(d.reason))) {
-                resolvedDesc = d;
-                break;
-            }
-        }
-        if (resolvedDesc == null) {
-            if (log.isWarnEnabled()) {
-                log.warn(String.format("Cannot find default rendition script for reason %s", reason));
-            }
-            return null;
-        }
+    public Rendition getDefaultRendition(DocumentModel doc, String reason, boolean store,
+            Map<String, Serializable> extendedInfos) {
         Map<String, Object> context = new HashMap<>();
         Map<String, Serializable> ei = extendedInfos == null ? Collections.emptyMap() : extendedInfos;
         NuxeoPrincipal currentUser = ClientLoginModule.getCurrentPrincipal();
         context.put("Document", doc);
         context.put("Infos", ei);
         context.put("CurrentUser", currentUser);
-        ScriptEngine engine = scriptEngineManager.getEngineByName(resolvedDesc.getScriptLanguage());
-        if (engine == null) {
-            throw new NuxeoException(
-                    "Engine not found for language: " + resolvedDesc.getScriptLanguage());
-        }
-        if (!(engine instanceof Invocable)) {
-            throw new NuxeoException("Engine " + engine.getClass().getName() + " not Invocable for language: "
-                    + resolvedDesc.getScriptLanguage());
-        }
-        Rendition rendition = null;
-        try {
-            engine.eval(resolvedDesc.getScript());
-            engine.getBindings(ScriptContext.ENGINE_SCOPE).putAll(context);
-            Object result = ((Invocable) engine).invokeFunction("run");
-            if (!(result instanceof String)) {
-                log.error("Failed to get rendition name (" + result + ")");
-            } else {
-                String defaultRenditionName = (String) result;
-                if (StringUtils.isBlank(defaultRenditionName)) {
-                    log.info("Default rendition name cannot be evaluated, returning null");
-                } else {
-                    try {
-                        rendition = getRendition(doc, defaultRenditionName);
-                    } catch (NuxeoException e) {
-                        log.error("Unable to use default rendition " + defaultRenditionName, e);
+        ScriptEngine engine;
+        for (int i = defaultRenditionDescriptors.size() - 1; i >= 0; i--) {
+            DefaultRenditionDescriptor desc = defaultRenditionDescriptors.get(i);
+            if ((StringUtils.isEmpty(reason) && StringUtils.isEmpty(desc.reason))
+                    || (reason != null && reason.equals(desc.reason))) {
+                engine = scriptEngineManager.getEngineByName(desc.getScriptLanguage());
```

Comment:
nit: can't we reuse the same engine and move this outside the loop? I guess it's safe to assume scripts won't mess up global context

Reply:
It it true that we just have `javacript` use case for now. But we can potentially have different script language. I'd say no.